v2/r/sample.R

```diff
@@ -0,0 +1,101 @@
+# Yelp API v2.0 code sample.
+# 
+# This program demonstrates the capability of the Yelp API version 2.0
+# by using the Search API to query for businesses by a search term and location.
+# 
+# Please refer to http://www.yelp.com/developers/documentation for the API documentation.
+# 
+# This program requires some R libraries including "httr", which you can install via:
+# `packages.install("httr")`
+# `packages.install("httpuv")`
+# etc.
+# 
+# Sample usage of the program:
+# `> source("sample.R")`
+# (output to the screen)
+# or
+# `R CMD BATCH sample.R`
+# (output to sample.Rout)
+
+# Required packages
+require(httr)
+require(httpuv)
+require(jsonlite)
+require(base64enc)
+
+# Your credentials, from https://www.yelp.com/developers/manage_api_keys
+consumerKey = ""
+consumerSecret = ""
+token = ""
+token_secret = ""
+
+yelp_query <- function(path, query_args) {
+  # Use OAuth to authorize your request.
+  myapp <- oauth_app("YELP", key=consumerKey, secret=consumerSecret)
+  sig <- sign_oauth1.0(myapp, token=token, token_secret=token_secret)
+
+  # Build Yelp API URL.
+  scheme <- "https"
+  host <- "api.yelp.com"
+  yelpurl <- paste0(scheme, "://", host, path)
+
+  # Make request.
+  results <- GET(yelpurl, sig, query=query_args)
+
+  # If status is not success, print some debugging output.
+  HTTP_SUCCESS <- 200
+  if (results$status != HTTP_SUCCESS) {
+      print(results)
+  }
+  return(results)
+}
+
+yelp_search <- function(term="dinner", location="Boston, MA", limit=3) {
```

Comment:
2 things:
* Seems weird to have a default value for search, but not for business. I understand why that things can happen to a business (closing, etc) so we don't want to default to anything, but it makes it inconsistent. Removing the default values also makes a clearer separation between demo code and a function someone could reuse.
* The [stackoverflow post](http://stackoverflow.com/questions/27728318/using-the-yelp-api-with-r-attempting-to-search-business-types-using-geo-coordin) that you referenced was originally about how to search with geo coordinates, why not make function that does this since we are writing the sample anyway. At risk of over engineering (this is a sample not a clientlib), you could make a [dumb function that accepts kwargs](https://www.quora.com/What-is-the-R-equivalent-of-Pythons-**kwargs) which simply builds and signs the url with the kwargs and make two separate functions, one that searches with a text location and one that searches with lat long.